File: westcoast/src/main/java/frc/motorFactory/MotorFactoryTalon.java
package frc.motorFactory;

import edu.wpi.first.wpilibj.SpeedController;
import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;

public class MotorFactoryTalon implements MotorFactory {
    public SpeedController create(int channelMain, int channel1, int channel2) {
        WPI_TalonSRX motorMain = new WPI_TalonSRX(channelMain);
        WPI_TalonSRX motor1 = new WPI_TalonSRX(channel1);
        WPI_TalonSRX motor2 = new WPI_TalonSRX(channel2);
 
        motor1.follow(motorMain);
        motor2.follow(motorMain);

        return motorMain;
    }    
}

File: westcoast/src/main/kotlin/frc/robot/motorfactory/MotorFactoryTalon.kt
package frc.robot.motorfactory

import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX
import edu.wpi.first.wpilibj.SpeedController

class MotorFactoryTalon : MotorFactory {
  override fun create(channelMain: Int, channel1: Int, channel2: Int): SpeedController {
    val motorMain = WPI_TalonSRX(channelMain)
    val motor1 = WPI_TalonSRX(channel1)
    val motor2 = WPI_TalonSRX(channel2)
    motor1.follow(motorMain)
    motor2.follow(motorMain)
    return motorMain
  }
}